sudachi/src/dic/character_category.rs

```diff
@@ -46,25 +42,23 @@ pub enum Error {
 struct Range {
     begin: u32,
     end: u32,
-    categories: CategoryTypes,
-}
-
-impl Range {
-    fn contains(&self, cp: u32) -> bool {
-        self.begin <= cp && cp < self.end
-    }
-    fn lower(&self, cp: u32) -> bool {
-        self.end <= cp
-    }
-    fn _higher(&self, cp: u32) -> bool {
-        cp < self.begin
-    }
+    categories: CategoryType,
 }
 
 /// CharacterCategory holds mapping from character to character category type
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Clone)]
 pub struct CharacterCategory {
-    ranges: Vec<Range>,
+    boundaries: Vec<u32>,
+    categories: Vec<CategoryType>,
```

Comment:
I feel it's better to have a comment about how boundaries and categories correspond to.
(we have `boundaries.len()+1` categories and `categories[i]` is for range `boundaries[i-1]..boundaries[i]`, right?)

Reply:
Certainly, will add.